### PR TITLE
Harpy Visual Rework

### PR DESCRIPTION
**Pull request by VMSolidus**
## About the PR
This is the third and final update to Harpies, after this PR is merged, they will finally be at a state I would consider to be finished. This PR functions as a complete visual rework to Harpies, adding several new visual features to them, including a complete rework of their layering mechanics. This alleviates several of the clipping issues they've had, and finally allows for me to create bespoke new species sprites for their clothing.  

In addition, @ps3moira and @stillxicarus have contributed entirely new sprites for use with Harpies. They now have new high-detail wing marking options, as well as fully avian limbs. 

## Why / Balance
Birds.

## Technical details
A new visual system is being added, for handling selectively hiding markings from Harpies when they equip a specific kind of outerwear(Hardsuits & Softsuits). This novel technique involves selectively toggling visibility to their wings and tail, while we can then replace their wing with that of a covered hardsuit variant when such clothing is equipped. It's likely that I could of course re-use this technique for Lamia and Drider, in order to address some issues they have had in the past. 

## Media
New multi-shape wing markings, enabling a wider variety of marking options. These now feature hueshift maps to allow the wings to be more colorful than ever before.
https://github.com/DeltaV-Station/Delta-v/assets/16548818/8c121e36-62c3-4660-b6fd-b61375924775
image
image

New Ultravision feature that allows Harpies to see in the same light spectrum as real life birds. 
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl: VMSolidus, PS3Moira, Stillxicarus
- add: Harpies have received a full visual rework. Featuring clothing sprites, new markings, a complete overhaul of their layering, as well as visual systems. 
- add: Harpies now have bird related speech verbs
- add: Two new traits have been added related to birds. Harpies now start by default with the new Ultraviolet Vision trait, letting them see a whole new spectrum of colors. This trait can be taken by other curious people. Harpies that wish to see in RGB colors can instead take the new Normal Vision trait.

### CODE REVIEW

**Comment by DebugOk** [comment]
I'm not sure how I feel about custom sprites for every jumpsuit, and I don't want that to be the standard. @leonardo-dabepis what are your thoughts on this? We already have vulps requiring custom sprites, and I'm not sure expanding that list is worth it. But I'm willing to let harpies be the only exception if you're okay with it

**Comment by leonardo-dabepis** [comment]
> I'm not sure how I feel about custom sprites for every jumpsuit, and I don't want that to be the standard. @leonardo-dabepis what are your thoughts on this? We already have vulps requiring custom sprites, and I'm not sure expanding that list is worth it. But I'm willing to let harpies be the only exception if you're okay with it

I think this is fine. The custom sprites look good on the harpy and the edits that would need to be made to accommodate new clothing items are minimal.

**Comment by VMSolidus** [comment]
image
So as with some new novel techniques I've come up with for the layering, it isn't 100% necessary for them to look good with existing sprites. However, they do look nicer with species variations. As in the example above, the Atmos tech right now has a custom jumpsuit sprite. Whereas the Mystagogue doesn't have one yet. 
image

They also don't require new sprites at all for jumpskirt variations. 
image

**Comment by VMSolidus** [comment]
image
image
But basically, the gist of how I accomplished this was that the marking layer for their wings is partially on the torso instead of the arms. Effectively, their wings are partially above and beneath clothing. This allowed me a previously impossible degree of flexibility in their appearance. These new techniques I learned are going to be useful for when we get around to bringing back Arachne, since I'll be able to put all these layering tech to good use for improving their sprites.

**Comment by VMSolidus** [comment]
This PR is now ready for review.

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml
Comment (DebugOk) [inline Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml:1]: Why is this adding slimes and reptiles back in?

**Review by DebugOk — CHANGES_REQUESTED** [review]
Didn't test, beyond my one comment I'll assume its fine

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml
Comment (VMSolidus) [reply Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml:1]: Looks like it came from a small error with resolving a merge conflict that happened earlier. I'll correct this, and then do a full QA pass on this PR before submitting approval. I suspect I have probably missed plenty of things like localization lines.

**Comment by VMSolidus** [comment]
My final QA pass has been completed. All Harpy markings have had a hueshift map applied to them for the sake of consistency, and all missing localizations have been added. There was also a missing marking state for the chest layer that I added in.

**Comment by DebugOk** [comment]
Please don't try to self merge without my review. Not that it would've worked as my request blocks it, and dismissing it would cause it to hang on the codeowner requirement

**Comment by VMSolidus** [comment]
Oh my bad, I had mistakenly though you had set an approval on it. I apologize for the misunderstanding.

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml
Comment (DebugOk) [inline Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml]: Whats going on here?

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml
Comment (DebugOk) [inline Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml]: And here

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml
Comment (DebugOk) [inline Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml]: And here

**Review by DebugOk — CHANGES_REQUESTED** [review]

**Comment by VMSolidus** [comment]
hueshifted birb
And here's the new look for the full body markings with hueshifts

File: Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml
Comment (VMSolidus) [reply Resources/Prototypes/Entities/Mobs/Customization/Markings/tattoos.yml]: Looks like more cases where this PR previously conflicted with another PR that added certain markings to more species. It conflicted with this PR at the time since it was allowing Harpies to use markings that in this PR would conflict with their new visuals. I've submitted a fix for these conflicts.

**Review by DebugOk — APPROVED** [review]
I dont like the fact this adds new sprites for clothing, but oh well

**Comment by Pharaz4** [comment]
Did _DV namespace not exist when harpies were added?

**Comment by Pharaz4** [comment]
Why didn't you put these textures under DeltaV @VMSolidus you made an opp with this pr
{04D0E68E-4A0C-4643-84AB-9B7BB6D57248}

**Pull request merged**
